### ISSUE TITLE
Use CapDashboard  with default options parameters not working 

### ISSUE THREAD

**Issue by evertonbzra** [dotnetcore/CAP#1428]
I updated my app with the latest version 7.2.2 and I was trying to use the CAP dashboard passing the default values on options parameters following the auth sample in the CAP repository.
At the verification step if the user is authenticated always return false even when the ChallengeAsync occurs.

Here's the cap dashboard configuration 

```cs
 cap.UseDashboard(d =>
 {
     d.UseChallengeOnAuth = true;
     d.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
     d.UseAuth = true;
     d.DefaultAuthenticationScheme = CookieAuthenticationDefaults.AuthenticationScheme;
 });
```
 
 Here's the AddAuthentication method configuration 
 
 ```cs
services
      .AddAuthentication(opt =>
      {
          opt.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
          opt.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
          opt.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
          
      })
      .AddJwtBearer(JwtBearerDefaults.AuthenticationScheme, opt =>
      {
          opt.Audience = "Audience";
          opt.Authority = "Jwt:Authority";
          opt.Events = new JwtBearerEvents
          {
              OnTokenValidated = async context => await BuybackJwtHandlers.HandleOnTokenValidated(context),
          };
      })
      .AddApiKey()
      .AddCookie(CookieAuthenticationDefaults.AuthenticationScheme)
      .AddOpenIdConnect(OpenIdConnectDefaults.AuthenticationScheme, opt =>
      {
          opt.Authority = "Authority";
          opt.ClientId = "client_Id";
          opt.ResponseType = "code";
          opt.UsePkce = true;

          opt.Scope.Clear();
          opt.Scope.Add("openid");
          opt.Scope.Add("profile");
          opt.Scope.Add("email");
          
          opt.SaveTokens = true;
      });
```

When I remove the lines  
`opt.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;`
 and 
`opt.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;`  
and 
include the `opt.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;` 
the dashboard works well as it's expected.

Why isn't the AddCapDashboard working when I provide the options parameters **DefaultChallengeScheme** and **DefaultAuthenticationScheme**? It appears that CAP is not recognizing the values I passed through the UseDashboard.

**Comment by yang-xiaodong** [comment]
Please see out example : https://github.com/dotnetcore/CAP/tree/master/samples/Sample.Dashboard.Auth

and to code of  [`Authentication`](https://github.com/dotnetcore/CAP/blob/master/src/DotNetCore.CAP.Dashboard/CAP.BuilderExtension.cs#L88)

**Comment by yang-xiaodong** [comment]
Hi @evertonbzra ,

It's confirmed as you said, but I haven't had much time to do in-depth research yet. Please let me know if you have further information about add `DefaultAuthenticateScheme` caused not working.

**Comment by mviegas** [comment]
Hi @yang-xiaodong and @evertonbzra! I've been closely watching this issue and I think that I have some findings.

The sample calls the default [ASP.NET authentication middleware](https://github.com/dotnet/aspnetcore/blob/49b81a392f5aa99728b7dc437b1a7fa89759f3ea/src/Security/Authentication/Core/src/AuthenticationMiddleware.cs#L13) via `app.UseAuthentication()`. If we check the linked source code, we see this middleware basically looks for the `DefaultAuthenticateScheme` in order to authenticate the user. This is the framework's default behavior.

Since the default in the failing example is the `JwtBearerDefaults.AuthenticationScheme`, the user will not be authenticated against this [scheme's handler](https://github.com/dotnet/aspnetcore/blob/49b81a392f5aa99728b7dc437b1a7fa89759f3ea/src/Security/Authentication/JwtBearer/src/JwtBearerHandler.cs#L20) - if you check at line 81 if it doesn't found the authorization header, it returns a `NoResult`. Therefore, the authentication will fail. 

However, [since CAP will keep trying to challenge the authentication every time the user is not authenticated](https://github.com/dotnetcore/CAP/blob/7ba52a8defa73f3866001dea6dd6af434574c3d2/src/DotNetCore.CAP.Dashboard/CAP.BuilderExtension.cs#L90), the application will enter a loop where:

1. It challenges the request against the options specified scheme and executes the authentication.
2. On the redirect to the application, it will always enter the authentication middleware and try to authenticate against the default scheme, due to the default behavior of auth middleware.
3. It fails, since Jwt is set as the default scheme and its handler confirms there's no Jwt received.
4. Since the user in HttpContext is not authenticated, we're back to step 1. Inception. 😅

I still have to think a bit more to validate this hypothesis, let me know if you agree with it. 

If valid, we can think about possible improvements to remove this ambiguity from CAP dashboard auth and its doc and make it leaner.

**Comment by yang-xiaodong** [comment]
Fixed in version 8.0.0
